src/openvic-simulation/map/Map.cpp

```diff
@@ -529,21 +536,26 @@ bool Map::load_map_images(fs::path const& province_path, fs::path const& terrain
 					);
 				}
 			}
+
 			province_shape_image[idx].index = Province::NULL_INDEX;
 
-		set_terrain:
-			const TerrainTypeMapping::index_t terrain = terrain_data[idx];
-			TerrainTypeMapping const* mapping = terrain_type_manager.get_terrain_type_mapping_for(terrain);
-			if (mapping != nullptr) {
-				if (province_shape_image[idx].index != Province::NULL_INDEX) {
-					terrain_type_pixels_list[province_shape_image[idx].index - 1][&mapping->get_type()]++;
+			set_terrain:
+				pixels_per_province[idx]++;
+				x_sum_per_province[idx] += x;
+				y_sum_per_province[idx] += y;
```

Comment:
```suggestion
				if (province_shape_image[idx].index != Province::NULL_INDEX) {
					const size_t array_index = province_shape_image[idx].index - 1;
					pixels_per_province[array_index]++;
					x_sum_per_province[array_index] += x;
					y_sum_per_province[array_index] += y;
				}
```
`idx` is not the province index here (it's the pixel index), what you need is `province_shape_image[idx].index`, and also to check that it isn't zero since the `set_terrain` code is run regardless of whether a non-zero province index is found or not.

Reply:
I corrected this and renamed the variables to prevent future confusion.

Reply:
The index may not have been set on the shape_image at that point.